osbenchmark/synthetic_data_generator/models.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import os
+from typing import Optional, Dict, List, Any, Union
+import re
+
+from pydantic import BaseModel, Field, field_validator
+
+GB_TO_BYTES = 1024 ** 3
+
+class SettingsConfig(BaseModel):
+    workers: Optional[int] = Field(default_factory=os.cpu_count)
+    max_file_size_gb: Optional[int] = 40
+    docs_per_chunk: Optional[int] = 10000
+
+    @field_validator('workers', 'max_file_size_gb', 'docs_per_chunk')
+    def validate_values_are_positive_integers(cls, v):
+        if v is not None and v <= 0:
+            raise ValueError('Value in Settings portion must be a positive integer.')
```

Comment:
Printing out the value of `v` will help the user track the reason for the error better.